src/ipips/ipip-0431.md

```diff
@@ -68,9 +76,43 @@ in the future.
 - Clients of trustless gateways can use the fields from the metadata as an attestation that they
 performed the retrieval from the given server.
 
-- The `len` field in the metadata block allows clients to verify whether they received all CAR
+- For example, the metadata block could include a `car_bytes` field, the byte length of the CAR stream (excluding the metadata block). This would allow clients to verify whether they received all CAR
 bytes, which provides a backward-compatible solution for the [CARv1 streaming problem](https://github.com/ipfs/specs/pull/332) until new CAR version is introduced.
 
+- As another example, the metadata object includes the `error` field, allowing the server to pass back additional information about why the response is an error, such as why the CAR stream was incomplete.
+
+- In the SPARK use case, retrieval clients would like to prove they have retrieved an entire file from a specific retrieval provder that has implemented the trustless gateway spec. The additional metadata block allows checksums and signatures to be passed along with the data, allowing the retrieval client to create a proof of correct retrieval. For SPARK, the metadata properties object SHOULD include the following fields:
+
+```json
+{
+  "car_bytes": {
+    "description": "The total byte length of the CAR stream (excluding the 0x00 byte and the metadata block)",
+    "type": "integer"
+  },
+  "data_bytes": {
+    "description": "Total byte length of blocks (excluding the 0x00 byte and the metadata block, but including duplicates when present)",
+    "type": "integer"
+  },
+  "block_count": {
+    "description": "Total number of blocks present in the CAR stream (excluding the 0x00 byte and the metadata block, but including duplicates when present)",
+    "type": "integer"
+  },
+  "b3checksum": {
+    "description": "A Blake3 hash (checksum) of the CAR stream (excluding the 0x00 byte and the metadata block)",
+    "type": "string"
+  },
+  "content_path": {
+    "description": "The url path in the request as executed by the gateway",
+    "type": "string"
+  },
+  "query_params": {
+    "description": "The query string in the request as executed by the gateway",
+    "type": "string"
+  }
```

Comment:
This will be more complex, based on the discussion in the spec. If we want to show a full example of what SPARK expects, then we need to decide how we want to represent "cid", "subpath", "dag params" and "car params".